governance/default.go

```diff
@@ -665,6 +666,14 @@ func getGovernanceCacheKey(num uint64) common.GovernanceCacheKey {
 
 // Store new governance data on DB. This updates Governance cache too.
 func (g *Governance) WriteGovernance(num uint64, data GovernanceSet, delta GovernanceSet) error {
+	g.idxCacheLock.RLock()
+	idx := make([]uint64, 0, len(g.idxCache))
+	copy(idx, g.idxCache)
+	g.idxCacheLock.RUnlock()
+
+	if len(idx) > 0 && num <= idx[len(idx)-1] {
```

Comment:
If len (idx) is 0, is it not processed?

Reply:
@nogang If len(idx) is 0, it means that the function is called in making a new governance object. Because `WriteGovernance` is used to store the vote result and also to store the information written in genesis.json, len(idx) can be 0. 

The check routine is to prevent writing already stored governance information again.